include/revng/Pipeline/Location.h

```diff
@@ -176,6 +172,18 @@ locationFromString(const Rank &, std::string_view String) {
   return Location<Rank>::fromString(String);
 }
 
+/// A helper interface for location conversion.
+///
+/// It discloses the static `convert` member in an easier-to-access fashion.
+template<typename ResultRank, typename InputRank>
+  requires(RankConvertibleTo<ResultRank, InputRank>)
+inline constexpr Location<ResultRank>
+convertLocation(const ResultRank &Result, const Location<InputRank> &Input) {
+  return Location<ResultRank>::convert(Input);
+}
```

Comment:
The argument is unused, this doesn't make much sense.

Reply:
(sorry for butting in into another person's PR, but)

It's exactly the same as the first argument of the function above it: this lets the caller to just pass `revng::ranks::Whatever` as an argument instead of trying to obtain its type (which is non-trivial) to pass as a template argument.

Reply:
I'd prefer to use `decltype` in the caller then having an unused argument

Reply:
Are you sure you'd rather go
```cpp
auto NewLocation = pipeline::convertLocation<decltype(revng::ranks::Function)>(MyLocation);
```
over
```cpp
auto NewLocation = pipeline::convertLocation(revng::ranks::Function, MyLocation);
```

I'd say the second one is a lot easier to reason about.

Reply:
Fair, it's a bit misleading for the user, but until he doesn't open the hood it's all good.